webofneeds/won-bot/src/main/java/won/bot/framework/eventbot/action/impl/mail/receive/MailCommandAction.java

```diff
@@ -0,0 +1,97 @@
+import won.bot.framework.eventbot.action.EventBotActionUtils;
+import won.bot.framework.eventbot.action.impl.mail.model.WonURI;
+import won.bot.framework.eventbot.action.impl.mail.receive.util.ActionType;
+import won.bot.framework.eventbot.bus.EventBus;
+import won.bot.framework.eventbot.event.Event;
+import won.bot.framework.eventbot.event.impl.mail.OpenConnectionEvent;
+import won.bot.framework.eventbot.event.impl.mail.CloseConnectionEvent;
+import won.bot.framework.eventbot.event.impl.mail.MailCommandEvent;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+import java.io.IOException;
+import java.net.URI;
+
+/**
+ * Created by fsuda on 18.10.2016.
+ */
+public class MailCommandAction  extends BaseEventBotAction {
+    private String mailIdUriRelationsName;
+
+    public MailCommandAction(EventListenerContext eventListenerContext, String mailIdUriRelationsName) {
+        super(eventListenerContext);
+        this.mailIdUriRelationsName = mailIdUriRelationsName;
+    }
+
+    @Override
+    protected void doRun(Event event) throws Exception {
+        if(event instanceof MailCommandEvent){
+            MimeMessage message = ((MailCommandEvent) event).getMessage();
+            EventBus bus = getEventListenerContext().getEventBus();
+
+            try{
+                String replyToMailId = getReplyToMailId(message);
+
+                WonURI wonUri = EventBotActionUtils.getWonURIForMailId(getEventListenerContext(), mailIdUriRelationsName, replyToMailId);
+                assert wonUri != null;
+
+                ActionType actionType = determineAction(message);
+                logger.debug("Executing " + actionType + " on uri: " + wonUri.getUri() + " of type " + wonUri.getType());
+
+                switch(actionType) {
+                    case CLOSE_CONNECTION:
+                        bus.publish(new CloseConnectionEvent(wonUri.getUri()));
+                        break;
+                    case CONNECT:
+                        bus.publish(new OpenConnectionEvent(wonUri.getUri()));
+                        break;
+                    case NO_ACTION:
+                    default:
+                        //INVALID COMMAND
+                        logger.error("No command was given or assumed");
+                        break;
+                }
+            }catch(Exception e){
+                logger.error("no reply mail was set or found");
+            }
+        }
+    }
+
+    public static boolean isCommandMail(MimeMessage message){
+        try{
+            return !StringUtil.isEmpty(getReplyToMailId(message));
+        }catch(MessagingException me){
+            return false;
+        }
+    }
+
+    public static String getReplyToMailId(MimeMessage message) throws MessagingException {
+        return message.getHeader("In-Reply-To")[0];
+    }
+
+    private static ActionType determineAction(MimeMessage message) {
+        try {
+            String messageContent = (String) message.getContent();
```

Comment:
Let's be a little more forgiving about the spelling - allow `Connect`, `connect`, and `CONNECT`, but allow only whitespace or punctuation after that.